### PR TITLE
feat: Implement marketing footer component

### PR DESCRIPTION
**Pull request by Trishu-Patel**

### CODE REVIEW

File: packages/ember/src/components/mktg/footer.gts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        align-items-center gx-3 gy-4 gx-md-4 my-0"
+    >
+      {{yield}}
+    </div>
+  </template>
+}
+
+export default class MarketingFooterComponent extends Component<MarketingFooterSignature> {
+  get classList() {
+    let classes = ['bg-primary', 'text-light', 'mt-auto'];
```
Comment (KeithClinard) [inline packages/ember/src/components/mktg/footer.gts:46]: I don't think we need the `classList` pattern here if the list of classes is static.
I'm of the opinion that we should just add these classes directly to `<footer class=...`

**Review by KeithClinard — CHANGES_REQUESTED** [review]

File: packages/ember-test-app/app/templates/marketing.hbs [hunk 2]
```diff
@@ -203,4 +203,31 @@
...
     </form>
   </div>
-</main>
\ No newline at end of file
+</main>
+
+<Mktg::Footer @hasHorizontalLine={{true}}>
+  <:nav>
+    <Button class="ms-2 rounded-pill btn-light">Get Started</Button>
+    <a href="" class="text-light">Pricing &amp; Plans</a>
```
Comment (lnicely) [inline packages/ember-test-app/app/templates/marketing.hbs:211]: Should we add these classes just so we have an example of not underlining the links? `link-underline link-underline-opacity-0`

File: packages/ember/src/components/mktg/footer.gts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          class="row row-cols-1 row-cols-md-auto justify-content-between align-items-center"
+        >
+          <FooterSection>
+            {{yield to="nav"}}
+          </FooterSection>
+          <FooterSection @isCollapsible={{false}}>
+            {{yield to="social-media"}}
+          </FooterSection>
+        </div>
+        {{#if this.hasHorizontalLine}}
```
Comment (TSenter) [inline packages/ember/src/components/mktg/footer.gts:62]: Could this be named something else, maybe `hasDivider` or just `divided`?

File: packages/ember/src/components/mktg/footer.gts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    >
+      {{yield}}
+    </div>
+  </template>
+}
+
+export default class MarketingFooterComponent extends Component<MarketingFooterSignature> {
+  get hasHorizontalLine() {
+    return this.args.hasHorizontalLine ?? false;
+  }
```
Comment (TSenter) [inline packages/ember/src/components/mktg/footer.gts:47]: Also, if the default value is falsey, you can remove the getter entirely and convert this to a template-only component.

File: packages/ember-test-app/tests/integration/components/mktg/footer-test.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    assert.dom('footer div hr').exists();
+  });
+
+  test('it contains a navigation bar', async function (assert) {
+    await render(hbs`<Mktg::Footer>
+      <:nav>
+        Nav Test
+      </:nav>
+    </Mktg::Footer>`);
```
Comment (TSenter) [inline packages/ember-test-app/tests/integration/components/mktg/footer-test.ts:26]: For writing multiline templates, there's a different format we're using:
```suggestion
    await render(hbs`
      <Mktg::Footer>
        <:nav>
          Nav Test
        </:nav>
      </Mktg::Footer>
    `);
```

File: packages/ember-test-app/app/styles/app.css [hunk 6]
```diff
@@ -1 +1,8 @@
 @import url("marketing-theme.css");
+
+/* The following tag ensures that the footer is always at the bottom of the page (no matter the size of the page content). */
+div[data-theme="marketing"] {
```
Comment (TSenter) [inline packages/ember-test-app/app/styles/app.css:4]: Can we add something a little more specific than any `div` with the `data-theme` of `marketing`? Perhaps a class or id. Also, this doesn't seem to be working on my machine.
image

**Review by TSenter — CHANGES_REQUESTED** [review]

File: packages/ember/src/components/mktg/footer.gts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  Element: HTMLElement;
+  Args: {
+    hasHorizontalLine?: boolean;
+  };
+  Blocks: {
+    nav: [];
+    'social-media': [];
+    brand: [];
+    legal: [];
+    default?: [];
```
Comment (TSenter) [inline packages/ember/src/components/mktg/footer.gts:24]: Also, you don't appear to have a default block
```suggestion
```

**Review by TSenter — CHANGES_REQUESTED** [review]

File: packages/ember/src/components/mktg/footer.gts (hunk 3 above)
Comment (TSenter) [reply packages/ember/src/components/mktg/footer.gts:61]: Talking with some others, let's go with `hasDivider`

Comment (Thomas-Jean) [reply packages/ember/src/components/mktg/footer.gts:61]: I would prefer `hasDivider` as it better hints that it is a boolean

File: packages/ember-test-app/app/styles/app.css (hunk 6 above)
Comment (Trishu-Patel) [reply packages/ember-test-app/app/styles/app.css:4]: Would it be possible to add this css as a part of the theme generation process? This insures that this css in not forgotten when we create a new add with nrg-ui

File: packages/ember-test-app/tests/integration/components/mktg/footer-test.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | components | mktg/footer', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<Mktg::Footer />`);
```
Comment (lnicely) [inline packages/ember-test-app/tests/integration/components/mktg/footer-test.ts:10]: Can you convert this to the new test format using `<template>` tags?

**Review by KeithClinard — DISMISSED** [review]

**Review by TSenter — APPROVED** [review]

**Pull request merged**